system_tests/bigquery.py

```diff
@@ -673,8 +670,58 @@ def test_create_table_nested_schema(self):
         retry_403(dataset.create)()
         self.to_delete.append(dataset)
 
-        table = dataset.table(table_name, schema=_load_json_schema())
+        schema = _load_json_schema()
+        table = dataset.table(table_name, schema=schema)
         table.create()
         self.to_delete.insert(0, table)
         self.assertTrue(table.exists())
         self.assertEqual(table.name, table_name)
+
+        to_insert = []
+        # Data is in "JSON Lines" format, see http://jsonlines.org/
+        json_filename = os.path.join(WHERE, 'bigquery_test_data.jsonl')
+        with open(json_filename) as rows_file:
+            for line in rows_file:
+                mapping = json.loads(line)
+                to_insert.append(
+                    tuple(mapping[field.name] for field in schema))
+
+        errors = table.insert_data(to_insert)
+        self.assertEqual(len(errors), 0)
+
+        retry = RetryResult(_has_rows, max_tries=8)
+        fetched = retry(self._fetch_single_page)(table)
+        self.assertEqual(len(fetched), len(to_insert))
+
+        for found, expected in zip(sorted(fetched), sorted(to_insert)):
+            self.assertEqual(found[0], expected[0])            # Name
+            self.assertEqual(found[1], int(expected[1]))       # Age
+            self.assertEqual(found[2], expected[2])            # Weight
+            self.assertEqual(found[3], expected[3])            # IsMagic
+
+            self.assertEqual(len(found[4]), len(expected[4]))  # Spells
+            for f_spell, e_spell in zip(found[4], expected[4]):
+                self.assertEqual(f_spell['Name'], e_spell['Name'])
+                parts = time.strptime(
+                    e_spell['LastUsed'], '%Y-%m-%d %H:%M:%S UTC')
+                e_used = datetime.datetime(*parts[0:6], tzinfo=UTC)
+                self.assertEqual(f_spell['LastUsed'], e_used)
+                self.assertEqual(f_spell['DiscoveredBy'],
+                                 e_spell['DiscoveredBy'])
+                self.assertEqual(f_spell['Properties'], e_spell['Properties'])
+
+                e_icon = base64.standard_b64decode(
+                    e_spell['Icon'].encode('ascii'))
+                self.assertEqual(f_spell['Icon'], e_icon)
```

Comment:
Why does this have to be so involved? Why can't we just say `found[4] == expected[4]`?